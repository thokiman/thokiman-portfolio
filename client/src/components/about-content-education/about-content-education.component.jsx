import React from "react";
import { CSSTransitionGroup } from "react-transition-group";
import { connect } from "react-redux";
import { createStructuredSelector } from "reselect";

import {
  selectEducationItemsLastBachelor,
  selectEducationItemsLastExperiences,
  selectEducationItemsLastCourses,
  selectPersonality,
} from "../../redux/about/about.selectors";
import AboutEducationBachelor from "../about-education-bachelor/about-education-bachelor.component";
import AboutEducationExperience from "../about-education-experience/about-education-experience.component";
import AboutEducationCourse from "../about-education-course/about-education-course.component";
import AboutEducationPersonality from "../about-education-personality/about-education-personality.component";
import { EducationContainer } from "./about-content-education.styles";
import { selectIsSideBarHidden } from "../../redux/header/header.selectors";

const AboutEducation = ({
  lastBachelor,
  lastExperiences,
  lastCourses,
  personality,
  isSideBarHidden,
}) => {
  return (
    <EducationContainer $issidebarhidden={isSideBarHidden}>
      <CSSTransitionGroup
        transitionAppear={true}
        transitionAppearTimeout={800}
        transitionEnterTimeout={600}
        transitionLeaveTimeout={200}
        transitionName={"slide-out"}
      >
        <AboutEducationBachelor lastBachelor={lastBachelor} />
        <AboutEducationExperience lastExperience={lastExperiences} />
        <AboutEducationCourse lastCourse={lastCourses} />
        <AboutEducationPersonality personality={personality} />
      </CSSTransitionGroup>
    </EducationContainer>
  );
};

const mapStateToProps = createStructuredSelector({
  lastBachelor: selectEducationItemsLastBachelor,
  lastExperiences: selectEducationItemsLastExperiences,
  lastCourses: selectEducationItemsLastCourses,
  personality: selectPersonality,
  isSideBarHidden: selectIsSideBarHidden,
});

export default connect(mapStateToProps)(AboutEducation);
