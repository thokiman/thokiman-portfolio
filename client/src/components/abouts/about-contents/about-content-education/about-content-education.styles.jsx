import styled, { css } from 'styled-components';

const backgroundEducationContainerColor = 'rgba(18, 129, 148, 0.8)';
const maxWidthSmartphone = 800;
const AboutContentEducationIsNotActivated = css`
  display: none;
`;
const getSideBarActive = ({ $issidebaractive }) => {
  if ($issidebaractive) return AboutContentEducationIsNotActivated;
  return null;
};
export const EducationContainer = styled.div`
  background-color: ${backgroundEducationContainerColor};
  width: 64.5vw;
  margin-right: 2vw;
  padding: 40px 0;
  height: 95%;
  overflow-y: scroll;
  overflow-x: hidden;
  font-family: baskerville-old-face;
  @media only screen and (max-width: ${maxWidthSmartphone}px) {
    width: 100%;
    margin-right: 0;
    ${getSideBarActive}
  }
`;

EducationContainer.displayName = 'EducationContainer';
