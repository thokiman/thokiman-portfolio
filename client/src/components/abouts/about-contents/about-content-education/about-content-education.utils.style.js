import {
  hasScreenSize1,
  hasScreenSize1_1,
  hasScreenSize1_2,
  hasScreenSize1_3,
  hasScreenSize2,
  hasScreenSize3_1,
  hasScreenSize3_2,
  hasScreenSize3_3,
  hasScreenSize3_4,
  hasScreenSize3_5,
  hasScreenSize3_6,
  hasScreenSize4,
  hasScreenSize5,
  hasScreenSizeWindow,
  hasScreenSizeWindow_1,
} from '../../../../utils/common.component.utils';

export const measureEducationContainerProps = (viewWidth, viewHeight) => {
  const educationContainerOffset = 5;
  let educationContainerProps;

  if (hasScreenSizeWindow(viewWidth, viewHeight)) {
    console.log('about-education', 1);
    educationContainerProps = {
      top: 0,
    };
  } else if (hasScreenSizeWindow_1(viewWidth, viewHeight)) {
    console.log('about-education', 2);
    educationContainerProps = {
      top: 0,
    };
  } else if (hasScreenSize1_1(viewWidth, viewHeight)) {
    console.log('about-education', 3);
    educationContainerProps = {
      top: 0,
    };
  } else if (hasScreenSize1_2(viewWidth, viewHeight)) {
    console.log('about-education', 4);
    educationContainerProps = {
      top: 0,
    };
  } else if (hasScreenSize1_3(viewWidth, viewHeight)) {
    console.log('about-education', 5);
    educationContainerProps = {
      top: 0,
    };
  } else if (hasScreenSize1(viewWidth)) {
    console.log('about-education', 6);
    educationContainerProps = {
      top: 0 + educationContainerOffset,
    };
  } else if (hasScreenSize2(viewWidth)) {
    console.log('about-education', 7);
    educationContainerProps = {
      top: 0 + educationContainerOffset,
    };
  } else if (hasScreenSize3_1(viewWidth, viewHeight)) {
    console.log('about-education', 8);
    educationContainerProps = {
      top: 0 + educationContainerOffset,
    };
  } else if (hasScreenSize3_2(viewWidth, viewHeight)) {
    console.log('about-education', 9);
    educationContainerProps = {
      top: 0 + educationContainerOffset,
    };
  } else if (hasScreenSize3_3(viewWidth, viewHeight)) {
    console.log('about-education', 10);
    educationContainerProps = {
      top: 0 + educationContainerOffset,
    };
  } else if (hasScreenSize3_4(viewWidth, viewHeight)) {
    console.log('about-education', 11);
    educationContainerProps = {
      top: 0 + educationContainerOffset,
    };
  } else if (hasScreenSize3_5(viewWidth, viewHeight)) {
    console.log('about-education', 12);
    educationContainerProps = {
      top: 0 + educationContainerOffset,
    };
  } else if (hasScreenSize3_6(viewWidth, viewHeight)) {
    console.log('about-education', 13);
    educationContainerProps = {
      top: 0 + educationContainerOffset,
    };
  } else if (hasScreenSize4(viewWidth, viewHeight)) {
    console.log('about-education', 14);
    educationContainerProps = {
      top: 0 + educationContainerOffset,
    };
  } else if (hasScreenSize5(viewWidth)) {
    console.log('about-education', 15);
    educationContainerProps = {
      top: 0 + educationContainerOffset,
    };
  } else {
    console.log('about-education', 16);
    educationContainerProps = {
      top: 20,
    };
  }

  return educationContainerProps;
};
