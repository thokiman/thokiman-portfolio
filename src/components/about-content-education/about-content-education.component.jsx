import React from "react";
import ReactCSSTransitionGroup from "react-addons-css-transition-group";
import { connect } from "react-redux";
import { createStructuredSelector } from "reselect";

import {
  selectEducationItemsLastBachelor,
  selectEducationItemsLastExperiences,
  selectEducationItemsLastCourses,
  selectPersonality,
} from "../../redux/about/about.selectors";
import AboutEducationBachelor from "../about-education-bachelor/about-education-bachelor.component";
import AboutEducationExperience from "../about-education-experience/about-education-experience.component";
import AboutEducationCourse from "../about-education-course/about-education-course.component";
import AboutEducationPersonality from "../about-education-personality/about-education-personality.component";
import "./about-content-education.styles.scss";

const AboutEducation = ({
  lastBachelor,
  lastExperiences,
  lastCourses,
  personality,
}) => {
  return (
    <div className="education-container">
      <ReactCSSTransitionGroup
        transitionAppear={true}
        transitionAppearTimeout={800}
        transitionEnterTimeout={600}
        transitionLeaveTimeout={200}
        transitionName={"slide-out"}
      >
        <AboutEducationBachelor lastBachelor={lastBachelor} />
        <AboutEducationExperience lastExperience={lastExperiences} />
        <AboutEducationCourse lastCourse={lastCourses} />
        <AboutEducationPersonality personality={personality} />
      </ReactCSSTransitionGroup>
    </div>
  );
};

const mapStateToProps = createStructuredSelector({
  lastBachelor: selectEducationItemsLastBachelor,
  lastExperiences: selectEducationItemsLastExperiences,
  lastCourses: selectEducationItemsLastCourses,
  personality: selectPersonality,
});

export default connect(mapStateToProps)(AboutEducation);
