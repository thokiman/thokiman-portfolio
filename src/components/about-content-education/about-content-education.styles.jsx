import styled from "styled-components";

export const EducationContainer = styled.div`
  background-color: rgba(18, 129, 148, 0.8);
  width: 64.5vw;
  margin-right: 2vw;
  padding: 40px 0;
  height: 95%;
  overflow-y: scroll;
  overflow-x: hidden;
  font-family: baskerville-old-face;
`;

EducationContainer.displayName = "EducationContainer";
