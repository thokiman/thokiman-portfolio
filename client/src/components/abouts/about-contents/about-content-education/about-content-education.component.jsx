import React from 'react';
import { connect } from 'react-redux';
import { createStructuredSelector } from 'reselect';
import {
  selectEducationItemsLastBachelor,
  selectEducationItemsLastCourses,
  selectEducationItemsLastExperiences,
  selectPersonality,
} from '../../../../redux/about/about.selectors';
import { selectIsSideBarActive } from '../../../../redux/header/header.selectors';
import AboutEducationBachelor from '../../about-educations/about-education-bachelor/about-education-bachelor.component';
import AboutEducationCourse from '../../about-educations/about-education-course/about-education-course.component';
import AboutEducationExperience from '../../about-educations/about-education-experience/about-education-experience.component';
import AboutEducationPersonality from '../../about-educations/about-education-personality/about-education-personality.component';
import { EducationContainer } from './about-content-education.styles';

export const AboutEducation = ({
  lastBachelor,
  lastExperiences,
  lastCourses,
  personality,
  isSideBarActive,
}) => {
  return (
    <EducationContainer
      className='education-container'
      $issidebaractive={isSideBarActive}
    >
      <AboutEducationBachelor lastBachelor={lastBachelor} />
      <AboutEducationExperience lastExperience={lastExperiences} />
      <AboutEducationCourse lastCourse={lastCourses} />
      <AboutEducationPersonality personality={personality} />
    </EducationContainer>
  );
};

const mapStateToProps = createStructuredSelector({
  lastBachelor: selectEducationItemsLastBachelor,
  lastExperiences: selectEducationItemsLastExperiences,
  lastCourses: selectEducationItemsLastCourses,
  personality: selectPersonality,
  isSideBarActive: selectIsSideBarActive,
});

export default connect(mapStateToProps)(AboutEducation);
