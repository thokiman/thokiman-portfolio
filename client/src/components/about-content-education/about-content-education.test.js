import React from "react";
import { shallow } from "enzyme";

import { AboutEducation } from "./about-content-education.component";

describe("AboutEducation component", () => {
  let wrapper;

  const mockArray = [];
  const mockFalseBool = false;
  beforeEach(() => {
    const mockProps = {
      lastBachelor: mockArray,
      lastExperiences: mockArray,
      lastCourses: mockArray,
      personality: mockArray,
      isSideBarHidden: mockFalseBool,
    };
    wrapper = shallow(<AboutEducation {...mockProps} />);
  });

  it("should render AboutEducation component", () => {
    expect(wrapper).toMatchSnapshot();
  });
});
