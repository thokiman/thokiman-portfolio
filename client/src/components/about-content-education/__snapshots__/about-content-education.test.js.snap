// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AboutEducation component should render AboutEducation component 1`] = `ShallowWrapper {}`;
